src/nlp/Preprocessing.py

```diff
@@ -253,3 +257,75 @@ def correct(self):
       corrected = doc._.outcome_spellCheck
 
     return corrected
+
+  def handleAbbreviations(self, abbrDatabase, type):
+    """
+      Performs automatic correction of abbreviations and returns corrected text
+      This method relies on a database of abbreviations located at:
+      src/nlp/data/abbreviations.xlsx
+      This database contains the most common abbreviations collected from literarture and
+      it provides for each abbreviation its corresponding full word(s); an abbreviation might
+      have multple words associated. In such case the full word that makes more sense given the
+      context is chosen (see findOptimalOption method)
+      @ In, abbrDatabase, pandas dataframe, dataframe containing library of abbreviations
+                                            and their correspoding full expression
+      @ In, type, string, type of abbreviation method ('spellcheck','hard','mixed')
+      @ Out, options, list, list of corrected text options
+    """
+    if type == 'spellcheck':
+      unknowns = self.getMisspelledWords()
+    elif type == 'hard' or type=='mixed':
+      unknowns = []
+      splitSent = self.text.split()
+      for word in splitSent:
+        if word in abbrDatabase['Abbreviation'].values.tolist():
```

Comment:
This is fairly inefficient because making a list and then searching it for each word is duplicating a lot of work.  It would be better to store the information in the abbrDatabase in something where you can directly search, such as a dictionary.  A locally more efficient way to do this is:
```python
      abbreviationSet = set(abbrDatabase['Abbreviation'].values)
      for word in splitSent:
        if word in abbreviationSet:
```

Reply:
good point, fixed